### DISCUSSION TITLE
If port forwarded changes each start, how do I define the port for docker containers? [protonvpn]

### DISCUSSION THREAD

**Discussion by ppriorfl** [qdm12/gluetun#1883] [Q&A]
First of all thanks so much for this work - I'm a monthly supporter and think this project is awesome.

I changed VPN to ProtonVPN from Mullvad as the later stopped supporting port forwarding.  However mullvad had fixed port forwarding - for a given server it was always the same port.

With proton, each time I start my portainer stack and glueton starts up it connects and gives me a port forward (I can see in the logs).  But that port **changes each time the docker starts**

Since in my glueton docker compose I have to define what ports go to what dockers [i.e. my glueton container says port forwarding is assigned from protonvpn as port 51822 so I forward that to qbittorrent custom port 48667 as below]:

  - 51822:48667/tcp #qbittorrent
  - 51822:48667/udp #qbittorrent

I'm not at all clear how I can do this.  If I stop my stack to update the port lines above, then when I restart it I get new ports and it's back to step 1.  I'm chasing my tail.

Must be an answer, but I'm not seeing it.  appreciate any help.  Do I need to run this separately in a different stack?  If so, how does my original media stack then reference it?

**Comment by AlbyGNinja** [comment]
I created a python script, based on snippets found online, which keeps up to date my qbittorrent `listen_port` and `embedded_tracker_port` with the same forwarded one on gluetun.
For start you have to create a new volume to your host binding `/tmp/gluetun` container folder to `/whereveryouwant` on the host
Next, you read the file `/whereveryouwant/forwarded_port` which contains the forwarded port assigned by ProtonVPN
Finally you update qbittorrent ports matching the same forwarded one.
Put the script in a cron - Job done.

**Reply by stowaway423** [reply]
Would you mind sharing your scripts and how you added the volume ? : )

**Reply by AlbyGNinja** [reply]
sorry for the late, you can check it here https://github.com/AlbyGNinja/port-sync
